Warning fix: uninitialized variable

mariamosn opened the pull request:
Initialized the `bytes` variable. The previous use without initializtion was likely the cause of the build warnings.
Related to issue #545 

Signed-off-by: Maria Moșneag <mariamosneag@gmail.com>

nderjung reviewed (changes requested):
Hi @mariamosn! Thanks or this contribution :)

Just a small change is necessary for it to get accepted.  Please can you rebase/reword your commit message so as to [follow our commit message style guide](https://unikraft.org/docs/contributing/coding-style/#commit-message-format):

```
lib/vfscore: Initialize `bytes` variable to `0` in syscall definition `pwritev`

This commit fixes an uninitialized variable which would otherwise throw build
warnings.

GitHub-Fixes: #545
Signed-off-by: Maria Moșneag <mariamosneag@gmail.com>
```

razvand reviewed (changes requested):
Hi, @mariamosn , thanks for this.

See @nderjung 's comments on commit messages.

Your solution doesn't really fix the issue. The actual issue is in the function [`do_pwritev`](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/main.c#L560). The `bytes` argument is not initialized in case of error. So the solution is adding:

```
        *bytes = -1
```

at line [`578`](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/main.c#L560), right before:

```
       return -error;
```

mariamosn commented:
Thank you, @nderjung and @razvand, for all the help and explanations!

razvand commented on lib/vfscore/main.c:
```diff
@@ -587,7 +588,7 @@ UK_SYSCALL_R_DEFINE(ssize_t, pwritev, int, fd, const struct iovec*, iov,
 {
 	struct vfscore_file *fp;
-	ssize_t bytes;
+	ssize_t bytes = 0;
```
Remove this, it's no longer required.

mariamosn committed:
lib/vfscore: Initialize bytes variable to -1 in syscall pwritev

This commit initializes the bytes variable to -1 in case of error in
the function do_pwritev. The uninitialized variable would otherwise
throw build warnings.

GitHub-Fixes: #545
Signed-off-by: Maria Mosneag <mariamosneag@gmail.com>

razvand reviewed (approved):
Thanks, @mariamosn . This works.

@nderjung , please test it and see if it's OK on your side. Once you give the approval I would add the `Reviewed-by` message on my side.

StefanJum reviewed (approved):
Looks good @mariamosn, thank you.
@razvand please add your `Reviewed-by:` tag so this can be merged.

Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

razvand reviewed (approved):
Reviewed-by: Razvan Deaconescu <razvand@unikraft.io>
